util.go

```diff
@@ -23,3 +29,41 @@ func isDuplicateEnvVar(e v1.EnvVar, envVars []v1.EnvVar) bool {
 	}
 	return false
 }
+
+// injectPodSpecEnvVars injects a slice of EnvVars into each PodSpec container
+func injectPodSpecEnvVars(podSpec v1.PodSpec, envVars []v1.EnvVar) v1.PodSpec {
+	containers := []v1.Container{}
+	for _, c := range podSpec.Containers {
+		c.Env = mergeEnvVars(c.Env, envVars)
+		containers = append(containers, c)
+	}
+	podSpec.Containers = containers
+	return podSpec
+}
+
+// newList creates a new list of resources to create
+func newList(objects ...runtime.Object) v1.List {
+	rawExtensions := []runtime.RawExtension{}
+	for _, o := range objects {
+		rawExtensions = append(rawExtensions, runtime.RawExtension{
+			Object: o,
+		})
+	}
+
+	return v1.List{
+		TypeMeta: unversioned.TypeMeta{
+			Kind:       "List",
+			APIVersion: "v1",
+		},
+		Items: rawExtensions,
+	}
+}
```

Comment:
For a pipe to kubectl, creating a list of resources should have no functional difference to concatenating multiple docs.  In the future, if we want to support writing back out to files, then I'm not sure it's desirable to alter the structure of the docs when injecting the vars.